src/Templates/SearchResultsProductTemplate.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * ProductSearchTemplate class.
+ *
+ * @internal
+ */
+class SearchResultsProductTemplate {
+
+	const SLUG        = 'search-results-product';
+	const TITLE       = 'Product Search Results';
+	const DESCRIPTION = 'Template used to display search results for products';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	protected function init() {
+		add_filter( 'search_template_hierarchy', array( $this, 'update_search_template_hierarchy' ), 10, 3 );
+		add_filter( 'get_block_templates', array( $this, 'set_template_info' ) );
+	}
+
+	/**
+	 * When the search is for products and it used a block theme the Product Search Template is rendered.
```

Comment:
Small rewording suggestion:
```suggestion
	 * When the search is for products and a block theme is active, render the Product Search Template.
```